### ISSUE TITLE
Support "Inspect Widget" command

### ISSUE THREAD

**Issue by xster** [Dart-Code/Dart-Code#1068]
'i' in the resident runner

**Comment by DanTup** [comment]
👍

@jacob314/@devoncarew Is there a way to jump to source when items are selected? I found this event:

```js
{
"type": "Event",
"kind": "Inspect",
"isolate": {
	"type": "@Isolate",
	"fixedId": true,
	"id": "isolates\/56008419",
	"name": "main.dart:main()",
	"number": "56008419"
},
"inspectee": {
	"type": "@Instance",
	"class": {
		"type": "@Class",
		"fixedId": true,
		"id": "classes\/1849",
		"name": "RenderParagraph"
	},
	"kind": "PlainInstance",
	"id": "objects\/5"
}
}
```

However I can't figure out how I can get a location of where this instance was created from this. I tried digging through the IntelliJ source, but couldn't trace it back (I found some stuff like `toDartVmServiceValueForSourceLocation` but it uses a `DiagnosticNode` so I'm not sure what's going on there).

**Comment by jacob314** [comment]
ext.flutter.inspector 
will let you get details on the object being inspected currently.

ext.flutter.inspector.getSelectedWidget
and
ext.flutter.inspector.getSummaryWidget
are probably the two most useful for you. Both will return a JSON object that among other things contains the source location of the widget if it is available.
https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/widgets/widget_inspector.dart

you will need to call
ext.flutter.inspector.setPubRootDirectories
to specify which directories you want to consider part of the local Flutter project to use getSummaryWidget.

**Comment by DanTup** [comment]
@jacob314 Thanks! I had a quick go, but I can't figure out what args I'm supposed to supply to `getSelectedWidget`. In the Dart code they're named `previousSelectionId` (which I think can be null) and `groupName`. The code seems to assume it gets one named `objectGroup` (I'm guessing that's the second one), but I can't figure out what I'm supposed to pass.

I tried digging through the IntelliJ plugin but it didn't seem to pass a second argument:

https://github.com/flutter/flutter-intellij/blob/e5e31c60c10fc5065356eb41575fc333ffaa1c33/src/io/flutter/inspector/InspectorService.java#L758

Any pointers? (I'm on Hangouts - dantup@ if easier to chat there)

**Comment by jacob314** [comment]
Grab me on hangouts tomorrow morning and we'll get it working for you.

previousSelectionId can be null.
objectGroup is an arbitrary string of your choosing which you use to control the lifetimes of objects referenced by ids returned by ext.flutter.inspect.
when you are done with the objects in the group call disposeGroup.

You will pass in a JSON map of the form 
```{"arg": value, "objectGroup": objectGroupName}``` to invoke all the InspectorService methods that take 2 args. For this case, pass in
```{"arg": null, "objectGroup": "your-group-name"}```

then later call
```disposeGroup`
with
```{"objectGroup" : "your-group-name"}```
when you no longer care about the selection

**Comment by DanTup** [comment]
Fixed in #1069.
